### PR TITLE
HTML: opener and discarded auxiliary browsing context

### PR DESCRIPTION
**Pull request by annevk**
For https://github.com/whatwg/html/pull/4379.

### CODE REVIEW

File: html/browsers/windows/embedded-opener-remove-frame.html [hunk 1]
```diff
@@ -15,32 +41,17 @@
...
-    frameW.opener = "immaterial";
-    assert_equals(openerGet(), null, "opener after setting it \"immaterial\" via directly invoking the getter");
-    const openerDescImmaterial = Object.getOwnPropertyDescriptor(frameW, "opener");
-    assert_equals(openerDescImmaterial.value, "immaterial");
-    assert_true(openerDescImmaterial.writable);
-    assert_true(openerDescImmaterial.enumerable);
-    assert_true(openerDescImmaterial.configurable);
+    testOpener(frameW, window, () => frame.remove());
 
     t.done();
```
Comment (domenic) [inline html/browsers/windows/embedded-opener-remove-frame.html]: Can use t.step_func_done for this one too, right?

**Review by domenic — APPROVED** [review]

File: html/browsers/windows/embedded-opener-remove-frame.html (hunk 1 above)
Comment (annevk) [reply html/browsers/windows/embedded-opener-remove-frame.html]: No, due to different load event behavior.

**Comment by annevk** [comment]
This probably should not be merged until https://github.com/whatwg/html/pull/4379#issuecomment-466351278 is resolved somehow.

File: html/browsers/windows/embedded-opener-remove-frame.html [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+
+  const openerDesc = Object.getOwnPropertyDescriptor(otherW, "opener"),
+        openerGet = openerDesc.get;
+
+  assert_equals(openerGet(), thisW, "opener before removal via directly invoking the getter");
+  discardOtherBC();
+  assert_equals(otherW.opener, null, "opener after removal");
+  assert_equals(openerGet(), null, "opener after removal via directly invoking the getter");
+
+  otherW.opener = null;
```
Comment (bzbarsky) [inline html/browsers/windows/embedded-opener-remove-frame.html]: It's a little weird to do this after the opener is already asserted to be null, but I guess we were already testing that.

**Review by bzbarsky — APPROVED** [review]
This seems fine, modulo sorting out the `close()` behavior.  I commented over in https://github.com/whatwg/html/pull/4379 about what's going on there in Firefox...

File: html/browsers/windows/embedded-opener-remove-frame.html (hunk 2 above)
Comment (annevk) [reply html/browsers/windows/embedded-opener-remove-frame.html]: Mainly to ensure the setter doesn't do anything weird. I could remove this if desired...

**Review by domenic — COMMENTED** [review]
New version LGTM again. It assumes that the spec will queue a task for discarding, which per IRC conversations sounds like it's a good assumption.

**Pull request merged**